storage/simple_tree.hpp

```diff
@@ -64,6 +68,44 @@ class SimpleTree
         it->Sort(false);
   }
 
+  /// \brief Checks all nodes in tree to find an equal one. If there're several equal nodes
+  /// returns the first found.
+  /// \returns a poiter item in the tree if found and nullptr otherwise.
+  /// @TODO The complexity of the method is O(n). But the structure (tree) is built on the start of the program
+  /// and then activly on run time. This method should work at constant time.
```

Comment:
While creating the hierarchical tree it's possible to create a hash table near it.
key - string name.
value - several numbers which let us to find the item in the tree.

For your information:
SimpleTree is used only for storage tree for the time being.
The size of the tree is about 1500 nodes (leaves and not leaves).
Maximum depth is 3.

But you're right. Generally speaking it's not constant time.